### ISSUE TITLE
Test Case: Admin/rapid - add Cost Tier Table

### ISSUE THREAD

**Issue by leobardoramon** [codetodesign/NYUAMS#725]
Server : DEV-prodebs20170930

Please add a Cost Tier Table to Admin/rapid so that the LGS can charge separate rates for example : (A) alumni , (B) students , (C) etc

This table would work in the same way as can be seen in Admin/wide/'Cost Tier" table ( with Cost Tier A and Cost Tier B ; currently used to separate 42" v. 58" wide paper )

see PDF here : https://drive.google.com/open?id=1bB0sn4MZtfSxpJ-NDJKLStZV-oktifI7

**Comment by barryross** [comment]
@leobardoramon Thank you for the description and detailed PDF.  In reviewing the current setup for Wide, I am having a bit of trouble making a direct comparison b/c it seems like the devices are also bound to either "a" or "b" in the case of Wide Format.  Is this correct?

Can you please explain how the device designation of "a" or "b" correlates to the size of the paper "a" or "b"? 

Thank you

**Comment by barryross** [comment]
@leobardoramon Also..at what point will the admin need to designate who is making the purchase...in the "New Order Wizard"?

**Comment by leobardoramon** [comment]
@buck0land a few questions 

regarding the UI of New Order Wizard : 

 it will be a drop down to select which group the client is in . is this okay ?

From coding standpoint , this would be easier to implement in Order Parameters panel ( where Service Type is selected ) . is this okay ?

regarding the UI of Admin/rapid : 

This table would live above the RP Device dropdown in Admin/rapid ( so that it's always there ) . is this okay ?

**Comment by barryross** [comment]
@buck0land Would the ability  to create user groups and assign % tiers which will then be applied to an entire rapid order, independent of device and/or material selection be a solution?

**Comment by NYU-AMS** [comment]
Yes to all the questions bellow.

regarding the UI of New Order Wizard :

it will be a drop down to select which group the client is in . is this okay ? 

From coding standpoint , this would be easier to implement in Order Parameters panel ( where Service Type is selected ) . is this okay ? 

regarding the UI of Admin/rapid :

This table would live above the RP Device dropdown in Admin/rapid ( so that it's always there ) . is this okay ?

**Comment by NYU-AMS** [comment]
The tiers will apply to the entire order.

@buck0land Would the ability to create user groups and assign % tiers which will then be applied to an entire rapid order, independent of device and/or material selection be a solution?

**Comment by barryross** [comment]
@buck0land Great!  Thank you for the answers, I'm pleased to hear this all works for you!

**Comment by barryross** [comment]
@buck0land @leobardoramon So, I'm thinking through how I may need to factor in the adjusted order amount into the system and curious how this may need to be reflected site-wide. 

It seems to me that is the discount/surcharge was added to each part in an order...the site-wide financials should remain accurate...basically the discount/surcharge would be reflected in the cost of parts, and orders.

I'm thinking we may want to add an additional column to the Orders table that would indicate which user group that order was associated with.  Would this be helpful?  

I'm picturing the case where two identical orders would have different costs if they were associated with different user tiers - e.g. order #1 was for an undergrad and order #2 was for an alumn.  If this information was bound to the order, the difference in cost would be easier to understand.

**Comment by NYU-AMS** [comment]
Good catch!   We will need the new column in orders. This data has to be tracked.

**Comment by leobardoramon** [comment]
@barryross , @buck0land 

 I'm wondering what this might looks like in different areas . Since there is already a University Status parameter for Client profile , how should this be treated ? see below : 

 - Would Clients/"University Status" and Orders/"University Status" be connected in any way ?

 - Do we want to give Orders/"University Status" a different name or code ?
for example Alumni = "A" or "AL" , Student = "B" or "STU" , Other = "C" or "US-3" . Although I've given these different options , I would like to keep the point of simplicity and utility ; over-complicating this feature with special codes/abbreviations could just add clutter/confusion/bugs

 - How would this look on Orders table ?
I initially thought some kind of code/abbreviation column before/after the Material column indicating the cost tier that has been applied . Can be something as simple as "a","b","c" .

 - How would this look on the Admin/rapid & Admin/tdscan pages ?
I think we already have a good idea of what this will look like ( since the idea was inspired by Admin/wide settings

 - Will this be added to Invoice PDF ?

 - Do we want the client to see this clearly stated on their invoice ?
 I think so ,,, it may help in case someone accidentally charges a student as an alumni , the client would then see this and be able to correct it VERSUS if a client is upset that they're not getting a cheaper price ( they can take it or leave it )

 - Is this a setting that we want to be able to edit after an order has already been created ?

**Comment by barryross** [comment]
@buck0land Thanks!

**Comment by barryross** [comment]
@leobardoramon great points/questions above.  

@leobardoramon @buck0land I think Leo's first question is an important one for us to address in order to answer the rest of the questions.  Now I'm thinking that perhaps we can simply use a client's current university status.  

If every order is associated with a client, and every client is already assigned to a university status, do we 

a) need the ability to manually assign a specific order to a "university status"? (like a dropdown in the New Order Wizard as previously suggested  ), or

b) On a per-order basis, the wizard could just look up the associated client's particular university status "behind the scenes" and calculate the cost of the parts (and total order) based off of the client's university status and the settings assigned in admin panel

**Comment by buck0land** [comment]
It’s A.

**Comment by barryross** [comment]
@buck0land great, thank you for letting me know.
